packages/core/src/sync/index.ts

```diff
@@ -1099,89 +1130,104 @@ export async function* getLocalEventGenerator(params: {
-          });
+    const initialChildAddresses = new Map<Factory, Map<Address, number>>();
+
+    for (const filter of params.sources.map(({ filter }) => filter)) {
+      switch (filter.type) {
+        case "log":
+          if (isAddressFactory(filter.address)) {
+            const childAddresses = await params.syncStore.getChildAddresses({
+              factory: filter.address,
+            });
 
-        params.common.logger.debug({
-          service: "sync",
-          msg: `Extracted ${events.length} '${params.network.name}' events for timestamp range [${decodeCheckpoint(cursor).blockTimestamp}, ${decodeCheckpoint(queryCursor).blockTimestamp}]`,
-        });
+            initialChildAddresses.set(filter.address, new Map(childAddresses));
+          }
+          break;
+
+        case "transaction":
+        case "transfer":
+        case "trace":
+          if (isAddressFactory(filter.fromAddress)) {
+            const childAddresses = await params.syncStore.getChildAddresses({
+              factory: filter.fromAddress,
+            });
 
-        estimateSeconds = estimate({
-          from: decodeCheckpoint(cursor).blockTimestamp,
-          to: decodeCheckpoint(queryCursor).blockTimestamp,
-          target: params.limit,
-          result: events.length,
-          min: 10,
-          max: 86_400,
-          prev: estimateSeconds,
-          maxIncrease: 1.08,
-        });
+            initialChildAddresses.set(
+              filter.fromAddress,
+              new Map(childAddresses),
+            );
+          }
 
-        params.common.logger.trace({
-          service: "sync",
-          msg: `Updated '${params.network.name}' extract query estimate to ${estimateSeconds} seconds`,
-        });
+          if (isAddressFactory(filter.toAddress)) {
+            const childAddresses = await params.syncStore.getChildAddresses({
+              factory: filter.toAddress,
+            });
 
-        consecutiveErrors = 0;
-        cursor = queryCursor;
-        yield { events, checkpoint: cursor };
-      } catch (error) {
-        if (params.common.shutdown.isKilled) {
-          throw error;
-        }
+            initialChildAddresses.set(
+              filter.toAddress,
+              new Map(childAddresses),
+            );
+          }
 
-        params.common.logger.warn({
-          service: "sync",
-          msg: `Failed '${params.network.name}' extract query for timestamp range [${decodeCheckpoint(cursor).blockTimestamp}, ${decodeCheckpoint(to).blockTimestamp}]`,
-          error: error as Error,
+          break;
+      }
+    }
+
+    while (cursor < Math.min(syncCursor, toBlock)) {
+      const { blockData, cursor: queryCursor } =
+        await params.syncStore.getEventBlockData({
+          filters: params.sources.map(({ filter }) => filter),
+          fromBlock: cursor,
+          toBlock: Math.min(syncCursor, toBlock),
+          chainId: params.network.chainId,
+          limit: params.limit,
```

Comment:
I'm a bit confused about how this limit maps to actual # of events. If the app has log filters, transaction filters, and trace filters, is it possible that we'd be pull 3*limit rows into memory?